integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/SparkOpenLineageExtensionVisitorWrapper.java

```diff
@@ -190,10 +199,28 @@ private Optional<ImmutablePair<Object, Method>> getMethod(
   }
 
   private static List<Object> init(String testExtensionProvider)
-      throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+      throws ClassNotFoundException, IOException, InstantiationException, IllegalAccessException {
     List<Object> objects = new ArrayList<>();
+    // The following sequence of operations must be preserved as is.
+    // We cannot use ResourceFinder or ServiceLoader to determine if there are any
+    // OpenLineageExtensionProvider implementations because doing so involves classloading
+    // machinery.
+    // As a result, there is no reliable way to bypass the entire mechanism, even if
+    // no OpenLineageExtensionProvider implementations are present.
+    ClassLoader currentThreadClassloader = Thread.currentThread().getContextClassLoader();
```

Comment:
why is it needed to instantiate `currentThreadClassloader` here and pass to another method? 
It is not used in this context at all, isn't it?

Reply:
fixed